helix-lock/src/main/java/org/apache/helix/lock/ZKHelixNonblockingLockInfo.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.helix.lock;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.helix.ZNRecord;
+
+
+public class ZKHelixNonblockingLockInfo<K extends ZKHelixNonblockingLockInfo.InfoKey, V extends String> implements LockInfo<K, V> {
+
+  public static final String DEFAULT_OWNER_TEXT = "";
+  public static final String DEFAULT_MESSAGE_TEXT = "";
+  public static final long DEFAULT_TIMEOUT_LONG = -1L;
+  public static final String DEFAULT_TIMEOUT_TEXT = String.valueOf(DEFAULT_TIMEOUT_LONG);
+  private Map<InfoKey, String> lockInfo;
+
+  public enum InfoKey {
+    OWNER, MESSAGE, TIMEOUT
+  }
+
+  /**
+   * Constructor of ZKHelixNonblockingLockInfo that set each field to default data
+   */
+  public ZKHelixNonblockingLockInfo() {
+    lockInfo = new HashMap<>();
+    lockInfo.put(InfoKey.OWNER, DEFAULT_OWNER_TEXT);
+    lockInfo.put(InfoKey.MESSAGE, DEFAULT_MESSAGE_TEXT);
+    lockInfo.put(InfoKey.TIMEOUT, DEFAULT_TIMEOUT_TEXT);
+  }
+
+  /**
+   * Construct a ZKHelixNonblockingLockInfo using a ZNRecord format of data
+   * @param znRecord A ZNRecord that contains lock information in its simple fields
+   */
+  public ZKHelixNonblockingLockInfo(ZNRecord znRecord) {
```

Comment:
As I commented above, there should be a default constructor that builds instance with strongly typed parameters. And the other 2 constructors can call that one inside.

Reply:
I added the strongly typed parameter constructor. But currently I'm not calling this method using the other 2 constructors since I made the LockInfo class extend HelixProperty, and I used the super(). With super(), a ZNRecord is already created so we only need to update the values in the ZNRecord.